anime_downloader/commands/ezdl.py

```diff
@@ -0,0 +1,202 @@
+    help="Specifiy the directory to download to")
+@click.option(
+    '--quality', '-q', type=click.Choice(['360p', '480p', '720p', '1080p']),
+    help='Specify the quality of episode. Default-720p')
+@click.option(
+    '--fallback-qualities', '-fq', cls=util.ClickListOption,
+    help='Specifiy the order of fallback qualities as a list.')
+@click.option(
+    '--force-download', '-f', is_flag=True,
+    help='Force downloads even if file exists')
+@click.option(
+    '--file-format', '-ff', default='{animeinfo_anime_title}/{animeinfo_anime_title}_{provider}_{ep_no}',
+    help='Format for how the files to be downloaded be named.',
+    metavar='FORMAT STRING'
+)
+@click.option(
+    '--provider',
+    help='The anime provider (website) for search.',
+    type=click.Choice(sitenames)
+)
+@click.option(
+    '--external-downloader', '-xd',
+    help='Use an external downloader command to download. '
+         'Use "{aria2}" to use aria2 as downloader. See github wiki.',
+    metavar='DOWNLOAD COMMAND'
+)
+@click.option(
+    '--chunk-size',
+    help='Chunk size for downloading in chunks(in MB). Use this if you '
+         'experience throttling.',
+    type=int
+)
+@click.option(
+    '--disable-ssl',
+    is_flag=True,
+    help='Disable verifying the SSL certificate, if flag is set'
+)
+@click.option(
+    '--choice', '-c',type=int,
+    help='Choice to start downloading given anime number '
+)
+@click.option(
+    '--fallback-providers', '-fp', cls=util.ClickListOption,
+    help='Specifiy the order of fallback providers as a list.')
+@click.option("--skip-fillers", is_flag=True, help="Skip downloading of fillers.")
+
+@click.pass_context
+def command(ctx, anime_url, episode_range, url, player, skip_download, quality,
+            force_download, download_dir, file_format, provider,
+            external_downloader, chunk_size, disable_ssl, fallback_qualities, choice, skip_fillers,
+            fallback_providers):
+
+    query = anime_url[:]
+    util.print_info(__version__)
+
+    fallback_providers.insert(0, provider)
+    # Eliminates duplicates while keeping order
+    providers = sorted(set(fallback_providers),key=fallback_providers.index)  
+    # TODO: flag for fallback providers
+
+    info = animeinfo.search_anilist(query)
+    episode_count = info.episodes - 1
+    # Interprets the episode range for use in a for loop.
+    # 1:3 -> for _episode in range(1, 4):
+    episode_range = util.parse_episode_range(episode_count, episode_range)
+    episode_range_split = episode_range.split(':')
+
+    # Stores the choices for each provider, to prevent re-prompting search.
+    choice_dict = {}
+
+    for episode_range in range(int(episode_range_split[0]), int(episode_range_split[-1])+1):
+        # Exits if all providers are skipped.
+        if [choice_dict[i] for i in choice_dict] == [0]*len(providers):
+            logger.info('All providers skipped, exiting')
+            exit()
+
+        for provider in providers:
```

Comment:
Why check provider for every episode?

Reply:
Let's say twist was missing only ep 4 on a show. To make only that episode fall back you'd either do this or run all the providers and compare results.